src/irr/asset/CGLIWriter.cpp

```diff
@@ -186,13 +195,18 @@ namespace irr
 					const auto layerData = texture.data(gliLayer, gliFace, region->imageSubresource.mipLevel);
 					const auto sourceData = ptrBeginningOfRegion + (layer * layerSize);
 
-					for (uint32_t yPos = 0; yPos < textureGliImgHeight; ++yPos)
+					for (size_t index = 0; index < pixelOrBlockCount; ++index)
+					{
+						const auto posXOfTexelOrBlock = static_cast<uint32_t>(index % size_t(imgBufferWidth / sourceTexelOrBlockStride));
+						const auto posYOfTexelOrBlock = static_cast<uint32_t>(index / size_t(imgBufferWidth / sourceTexelOrBlockStride));
+
 						memcpy
 						(
-							reinterpret_cast<uint8_t*>(layerData) + (yPos * textureGliStrideInBytes),	// copy to a beginning of a certain row
-							sourceData + (yPos * imgBufferWidthInBytes),								// get adjusted beginning of a certain row using to copying process
-							imgBufferWidth																// get row stride with no pitch
+							reinterpret_cast<uint8_t*>(layerData) + (posYOfTexelOrBlock * textureGliStrideInBytes),	// copy to a beginning of a certain row
+							sourceData + (posYOfTexelOrBlock * imgBufferWidthInBytes),								// get adjusted beginning of a certain row using to copying process
+							textureGliStrideInBytes																    // get gli row stride with no pitch
```

Comment:
this need to be a minimum of the range between region offset and max coord (extent) and GLIByteStride - region offset 

Obviously with all values converted to bytes, and 3 checks whether all parameters that dea with dimensions and extents are aligned to block size before you even enter the loop.